lib/api/core/models/security/profile.js

```diff
@@ -78,84 +77,80 @@ class Profile {
+    }
+
     if (!Array.isArray(this.policies)) {
-      return Bluebird.reject(errorsManager.get('missing_mandatory_policies_attribute'));
+      return assertionError.reject('invalid_type', 'policies', 'object[]');
     }
 
     if (this.policies.length === 0) {
-      return Bluebird.reject(errorsManager.get('empty_policies_attribute'));
+      return assertionError.reject('empty_argument', 'policies');
     }
 
     let i = 0;
     for (const policy of this.policies) {
       if (!policy.roleId) {
-        return Bluebird.reject(errorsManager.get(
-          'missing_mandatory_roleId_attribute',
-          i));
+        return assertionError.reject('missing_argument', `policies[${i}].roleId`);
       }
 
       for (const member of Object.keys(policy)) {
         if (member !== 'roleId' && member !== 'restrictedTo') {
-          return Bluebird.reject(errorsManager.get(
-            'unexpected_attribute_in_policies',
-            i,
-            member));
+          return assertionError.reject(
+            'unexpected_argument',
+            `policies[${i}].${member}`,
+            '"roleId", "restrictedTo"');
         }
       }
 
       if (policy.restrictedTo) {
         if (!Array.isArray(policy.restrictedTo)) {
-          return Bluebird.reject(errorsManager.get(
-            'attribute_restrictedTo_not_an_array_of_objects',
-            i));
+          return assertionError.reject(
+            'invalid_type',
+            `policies[${i}].restrictedTo`,
+            'object[]');
         }
 
         let j = 0;
         for (const restriction of policy.restrictedTo) {
           if (restriction === null || typeof restriction !== 'object' || Array.isArray(restriction)) {
-            return Bluebird.reject(errorsManager.get(
-              'restrictedTo_field_must_be_an_object',
-              i,
-              j));
+            return assertionError.reject(
+              'invalid_type',
+              `policies[${i}].restrictedTo[${restriction}]`,
+              'object');
           }
 
           if (restriction.index === null || restriction.index === undefined) {
-            return Bluebird.reject(errorsManager.get(
-              'missing_mandatory_index_attribute_in_restrictedTo_array',
-              i,
-              j));
+            return assertionError.reject(
+              'missing_argument',
+              `policies[${i}].restrictedTo[${j}].index`);
           }
 
           if (typeof restriction.index !== 'string' || restriction.index.length === 0) {
-            return Bluebird.reject(errorsManager.get(
-              'index_attribute_is_empty_string',
-              i,
-              j));
+            return assertionError.reject(
+              'invalid_type',
+              `policies[${i}].restrictedTo[${j}].index`,
+              'string');
           }
 
           if (restriction.collections !== undefined && restriction.collections !== null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the nested one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ACollapsibleIfStatements)